src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyout.iOS8.iOS.cs

```diff
@@ -11,12 +11,42 @@
 
 namespace Windows.UI.Xaml.Controls
 {
-    public partial class MenuFlyout
-    {
+	public partial class MenuFlyout
+	{
 		private static DependencyProperty IsDestructiveProperty = ToolkitHelper.GetProperty("Uno.UI.Toolkit.MenuFlyoutItemExtensions", "IsDestructive");
 
 		private UIAlertController _alertController;
 
+		internal override UIView NativeTarget
+		{
+			get
+			{
+				if (Target is AppBarButton appBarButton
+					&& appBarButton.Superview == null
+					&& appBarButton.Parent is CommandBar commandBar
+					&& GetNativeAppBarButton(appBarButton) is { } nativeAppBarButton
+					&& commandBar.GetRenderer(() => new CommandBarRenderer(commandBar)).Native is { } navigationBar)
+				{
+					// If the AppBarButton is a proxy for a native navigation bar button, find the corresponding native view.
+
+					if (nativeAppBarButton.Image is { } image)
+					{
+						return navigationBar.FindSubviewsOfType<UIImageView>().FirstOrDefault(v => v.Image == image);
+					}
+					else if (!nativeAppBarButton.Title.IsNullOrEmpty())
+					{
+						return navigationBar.FindSubviewsOfType<UIButton>()
+							.SelectMany(b => b.FindSubviewsOfType<UILabel>())
+							.FirstOrDefault(l => l.Text == nativeAppBarButton.Title);
```

Comment:
Does this work if the button is only an icon?

Reply:
that's the case above, if AppBarButton.Icon is set then we set the native Image property: https://github.com/unoplatform/uno/blob/master/src/Uno.UI/Controls/AppBarButtonRenderer.iOS.cs#L89

I'll add a comment here to clarify ;)